CMakeLists.txt

```diff
@@ -32,6 +32,74 @@ set(CMAKE_JAVA_COMPILE_FLAGS -source 1.8 -target 1.8 -encoding utf-8)
 add_definitions(-DDUCKDB_BUILD_LIBRARY)
 
 if(MSVC)
+  remove_definitions(-DDUCKDB_EXTENSION_JEMALLOC_LINKED)
```

Comment:
Instead of removing them can we perhaps not add them in the first place? That seems like it would be cleaner. Or is the above list of files auto-generated?

Reply:
Yes, these definitions and the sources list is auto-generated by `vendor.py`, regarding jemalloc it contains the [following logic](https://github.com/duckdb/duckdb-java/blob/a179e869e51f43fda865cc37459cae8dad533fe2/vendor.py#L25). I cannot make sense of it, because these platform checks are done when the `vendor.py` itself is being run (how is it supposed to check for Android and WASM at that time, why checking for WASM for JDBC sources etc - it was added in #133, I asked Laurens in Discord about it). I assume these checks for jemalloc need to be run at CMake configure time instead. And the list of sources needs to be added by `vendor.py` into a separate CMake variable. But for now, without touching `vendor.py` this "filter out on Windows" logic looks the easiest.

Reply:
That makes sense. It looks like these checks were copied from the Python build, which indeed doesn't work because the vendor script does not get run per configuration. The correct checks should be in the CMake. I would copy over the checks [from here](https://github.com/duckdb/duckdb/blob/main/extension/extension_config.cmake#L16).